### DISCUSSION TITLE
initramfs unpacking failed: uncompression error

### DISCUSSION THREAD

**Discussion by thewhitestig** [ipxe/ipxe#662] [Q&A] [answered]
Hi folks,

I had been booting up CentOS7.x with undionly.kpxe for many years, now upgrading to Rocky Linux 8.5 and run into problem while unpacking initramfs.   Image file transfer was okay, just failed when uncompressing it.  The new initrd0.img is a lot bigger (1.1G vs 520M), am I hitting an image size limit of iPXE?  System memsize is 64G.

```
http://192.136.251.249/images/linux/vm-rl8.5/vmlinuz0...  0%      0%     61%     ok
http://192.136.251.249/images/linux/vm-rl8.5/initrd0.img...  1%      2%      5%      7%      8%      9%     10%     11%     13%     14%     17%     18%     20%     21%     22%     24%     26%     28%     29%     31%     32%     34%     36%     37%     38%     38%     41%     44%     46%     49%     54%     58%     61%     66%     67%     68%     70%     70%     70%     73%     75%     76%     79%     80%     82%     85%     86%     87%     89%     90%     91%     92%     92%     94%     95%     97%     98%     98%     ok
<snip>
[    4.080813] Unpacking initramfs...
[    4.527714] Initramfs unpacking failed: uncompression error
[    4.667206] Freeing initrd memory: 1057460K
```

Thanks,
The White Stig

**Comment by stappersg** [comment]
On Thu, Jun 16, 2022 at 06:25:15PM -0700, The White Stig wrote:
>  ...  The new initrd0.img is a lot bigger (1.1G vs 520M),
> am I hitting an image size limit of iPXE?
> Is there any way I can work around this?
> ```
> [    4.080813] Unpacking initramfs...
> [    4.527714] Initramfs unpacking failed: uncompression error
> [    4.667206] Freeing initrd memory: 1057460K
> ```
> 

Information I mis:
* CPU architecture where this happens
* Which kind of "Boot ROM environment" is in play
* Amont of memory the booting system has
* Size of the kernel ( initrd0.img is about 1.1Gbyte)
* Which network protocols are being used to transfer kernel and initrd
* Output of `imgstat` before iPXE executes `boot`
* Kernel output after `[    4.667206] Freeing initrd memory: 1057460K`

**Reply by thewhitestig** [reply]
Thanks for the reply, here are the info I gathered.
* CPU architecture where this happens
` x86_64   Intel(R) Xeon(R) Silver 4310 CPU @ 2.10GHz`
* Which kind of "Boot ROM environment" is in play
  ` It's an HP Proliant system we setup to network boot from its X710 NIC card with legacy BIOS, if that answer your question.`
* Amont of memory the booting system has
  ` 64GB`
* Size of the kernel ( initrd0.img is about 1.1Gbyte)
  `kernel (vmlinuz0) is 9.8M`
* Which network protocols are being used to transfer kernel and initrd
  `http for both`
* Output of `imgstat` before iPXE executes `boot`
  `vmlinuz0 : 10215728 bytes [bzImage] [SELECTED] 
"initrd=initrd0.img rootfstype=iso9660 rootflags=loop root=live:/vm-rl8.5.iso pcie_aspm=off console=ttyS0,115200 ip=192.136.251.3 gateway=192.136.251.0 netmask=255.255.0.0 dns=xxx.xxx.xxx.xxx hostname=hyp-1 timezone=CET BOOTIF=68:05:CA:E0:CE:68"
initrd0.img : 1082839015 bytes`
* Kernel output after `[    4.667206] Freeing initrd memory: 1057460K`
  `I've captred them in this file: [pxe-rl85.txt](https://github.com/ipxe/ipxe/files/8931217/pxe-rl85.txt)
`

**Reply by stappersg** [reply]
The `imgstat` output looks strange  and I'm not convinenced that _iPXE_ could be the culprit.
```text
 [    5.579530] md: If you don't use raid, use raid=noautodetect
 [    5.585507] md: Autodetecting RAID arrays.
 [    5.589624] md: autorun ...
 [    5.592424] md: ... autorun DONE.
 [    5.595992] VFS: Cannot open root device "live:/vm-rl8.5.iso" or unknown-block(0,0): error -19
 [    5.604635] Please append a correct "root=" boot option; here are the available partitions:
 [    5.613016] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
 [    5.621308] CPU: 22 PID: 1 Comm: swapper/0 Not tainted 4.18.0-348.20.1.el8_5.x86_64 #1
 [    5.629249] Hardware name: HPE ProLiant DL360 Gen10 Plus/ProLiant DL360 Gen10 Plus, BIOS U46 01/13/2022
 [    5.638672] Call Trace:
 [    5.641129]  dump_stack+0x5c/0x80
 [    5.644454]  panic+0xe7/0x2a9
 [    5.647434]  mount_block_root+0x24a/0x2e2
 [    5.651456]  ? do_early_param+0x91/0x91
 [    5.655300]  prepare_namespace+0x135/0x16b
 [    5.656449] usb 2-3: New USB device found, idVendor=0424, idProduct=2660, bcdDevice= 8.01
 [    5.659407]  kernel_init_freeable+0x1fe/0x228
 [    5.659410]  ? rest_init+0xaa/0xaa
```

**Reply by oschroeder8** [reply]
Try removing the `rootfstype`, `rootflags`, and `root` boot arguments from your iPXE file. I am booting Rocky 8.x via iPXE and I don't use those arguments.

**Reply by thewhitestig** [reply]
I tried removing those boot arguments, and added raid=noautodetect as well, but still failing to boot.  Is there any things else I can look for?
[pxe-rl85_new.txt](https://github.com/ipxe/ipxe/files/8943788/pxe-rl85_new.txt)

**Reply by stappersg** [reply]
Snippet from https://github.com/ipxe/ipxe/files/8943788/pxe-rl85_new.txt
```text
[    5.691207] No filesystem could mount root, tried: 
[    5.691207] 
[    5.697591] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
```

> Is there any things else I can look for?

[Yes](http://www.catb.org/~esr/faqs/smart-questions.html)

**Reply by thewhitestig** [reply]
I understand the kernel panic is caused by missing root fs, which is likely a result of failing to unpack initramfs.  This brings back us back to original question, what're the possibilities that would cause this "initramfs npacking failed: uncompression error"?  The same rocky linux 8.5 image and network boot setup works on another hardware.

**Reply by stappersg** [reply]
Acknowlege on
>  The same rocky linux 8.5 image and network boot setup works on another hardware.

**Reply by NiKiZe** [reply]
Ensure the initrd is valid, verify that the loaded file is correct size in iPXE using imgstat. And finally make sure that the kernel has decompression algorithms enabled.

Failed decompression is not always a problem, trailing garbage is common and can be ignored, but of course that should be verified.

**Comment by thewhitestig** [comment, marked as answer]
just a quick follow up on this, I've reduced the size of my rockylinux image and problem is solved.
